### PR TITLE
Add SpringBoot Tests

### PR DESCRIPTION
**Pull request by LuisL02**
# Pull Request Description: Add SpringBoot Tests

## Motivation and Context

This pull request addresses the need for improved test coverage and reliability in our SpringBoot-based application. As the project has grown, ensuring the stability of new and existing features has become increasingly important. By introducing comprehensive SpringBoot tests, we aim to:

- Detect regressions early in the development cycle
- Validate core business logic and API endpoints
- Facilitate safer refactoring and feature additions
- Provide confidence during deployments and releases

The motivation is also to align with industry best practices and team goals for continuous integration and delivery (CI/CD), where automated testing is a critical component.

## Description

This PR introduces a suite of automated tests using the SpringBoot testing framework. The changes include:

- **Test Classes and Methods:** New test classes have been added for major controllers, services, and repositories. Each test class covers various scenarios, including standard behavior, error handling, and edge cases.
- **Integration Testing:** Several integration tests verify the behavior of the application when multiple components interact, particularly focusing on the REST API endpoints.
- **Mocking and Test Utilities:** The PR employs mocking frameworks (e.g., Mockito) to isolate unit tests where appropriate, and utility classes have been included to facilitate data setup and assertions.
- **Test Configuration:** Application context and profiles for testing are configured to avoid conflicts with production data and settings.
- **Documentation:** Where necessary, comments and documentation have been added or updated to guide future contributors on how to run and extend the tests.

## Steps for Testing

To test the changes introduced in this PR:

1. **Checkout the PR branch:**
   ```bash
   git fetch origin pull/67/head:add-springboot-tests
   git checkout add-springboot-tests
   ```

2. **Ensure all dependencies are installed:**
   Make sure you have Java and Maven installed, as required by the project.

3. **Run all tests:**
   ```bash
   # For Maven projects:
   mvn clean test
   ```

4. **Review test output:**
   - All tests should pass.
   - Review code coverage reports if available (e.g., via Jacoco plugins).
   - Check logs for any warnings or errors in the testing phase.

5. **Manual verification (optional):**
   - Deploy the application to a local environment and verify key endpoints and flows, especially those with new or updated tests.

## Review Progress

- **Code Review:** The new tests and supporting code have been structured to be readable, maintainable, and aligned with team conventions. Please pay particular attention to:
  - Test coverage: Are all major features and edge cases tested?
  - Clarity and isolation: Are tests focused and independent, or do they have unnecessary dependencies?
  - Maintainability: Is it easy to extend tests for future features?

- **Feedback:** Reviewers are encouraged to suggest improvements to test cases, naming, and structure.

- **CI Status:** The PR will trigger the CI pipeline. All tests must pass before merging.

- **Next Steps:** Once approved and merged, future pull requests should continue to provide or update tests for any modified or newly added functionality.

---

**Thank you for reviewing this contribution to our test suite! Your feedback is appreciated.**

### CODE REVIEW

File: server/authentication-service/src/main/java/devops25/releaserangers/authentication_service/service/UserService.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        return userRepository.save(user);
+    }
+
+    public void deleteUser(String email) {
+        final User user = userRepository.findByEmail(email);
+        if (user == null) {
+            throw new IllegalArgumentException("User not found");
+        }
+        userRepository.delete(user);
+    }
```
[the remark this answers is not in the document]
Comment (LuisL02) [reply server/authentication-service/src/main/java/devops25/releaserangers/authentication_service/service/UserService.java:48]: Removed, since we would throw an exception in the controller in case the user would not be present

File: server/authentication-service/src/main/java/devops25/releaserangers/authentication_service/service/UserService.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            user.setEmail(update.getEmail());
+        }
+        if (update.getName() != null) {
+            user.setName(update.getName());
+        }
+        if (update.getPassword() != null) {
+            user.setPassword(update.getPassword());
+        }
+        return userRepository.save(user);
+    }
```
[the remark this answers is not in the document]
Comment (LuisL02) [reply server/authentication-service/src/main/java/devops25/releaserangers/authentication_service/service/UserService.java:40]: I did not want to move the entire logic one level down to the service, since it was already set up and running in the controller. However if one attribute is null in the controller it will still be null here, so we have to check again
